app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java

```diff
@@ -197,28 +234,45 @@ private void loadHeader(int position) {
-                        switch (adapter.getCurrentTableScale().get()) {
-                            case DEFAULT:
-                                scaleImage.setImageDrawable(AppCompatResources.getDrawable(view.getContext(), R.drawable.ic_zoomx3));
-                                break;
-                            case LARGE:
-                                scaleImage.setImageDrawable(AppCompatResources.getDrawable(view.getContext(), R.drawable.ic_zoomx1));
-                                break;
-                            case SMALL:
-                                scaleImage.setImageDrawable(AppCompatResources.getDrawable(view.getContext(), R.drawable.ic_zoomx2));
-                                break;
+
+            cornerView.findViewById(R.id.buttonRowScaleAdd).setOnClickListener(view -> {
+                        for (int i = 0; i < binding.tableLayout.getChildCount(); i++) {
+                            if (binding.tableLayout.getChildAt(i) instanceof TableView) {
+                                TableView table = (TableView) binding.tableLayout.getChildAt(i);
+                                DataSetTableAdapter adapter = (DataSetTableAdapter) table.getAdapter();
+                                adapter.scaleRowWidth(true);
+                                ViewGroup.LayoutParams params = cornerView.getLayoutParams();
+                                params.width = adapter.getRowHeaderWidth();
+                                cornerView.setLayoutParams(params);
+                                if (i == 0) {
+                                    presenterFragment.saveCurrentSectionMeasures(
+                                            adapter.getRowHeaderWidth(),
+                                            adapter.getColumnHeaderHeight()
+                                    );
+                                }
+                            }
                         }
-                        DataSetTableAdapter.TableScale tableScale = adapter.scale();
-                        analyticsHelper().setEvent(LEVEL_ZOOM, tableScale.name(), ZOOM_TABLE);
                     }
-                }
-            });
+            );
+            cornerView.findViewById(R.id.buttonRowScaleMinus).setOnClickListener(view -> {
+                        for (int i = 0; i < binding.tableLayout.getChildCount(); i++) {
+                            if (binding.tableLayout.getChildAt(i) instanceof TableView) {
+                                TableView table = (TableView) binding.tableLayout.getChildAt(i);
+                                DataSetTableAdapter adapter = (DataSetTableAdapter) table.getAdapter();
+                                adapter.scaleRowWidth(false);
+                                ViewGroup.LayoutParams params = cornerView.getLayoutParams();
+                                params.width = adapter.getRowHeaderWidth();
+                                cornerView.setLayoutParams(params);
+                                if (i == 0) {
```

Comment:
what does represent i==0?

Reply:
is 0 the header?

Reply:
A way to save the result just once in the loop

Reply:
why don't you save outside the loop?